packages/@react-spectrum/menu/stories/SubMenu.stories.tsx

```diff
@@ -0,0 +1,587 @@
+import {ContextualHelpTrigger, Item, Menu, MenuTrigger, Section, SubMenuTrigger} from '../';
+import defaultConfig, {render as renderMenuTrigger} from './MenuTrigger.stories';
+import {Dialog} from '@react-spectrum/dialog';
+import {Heading, Keyboard, Text} from '@react-spectrum/text';
+import React from 'react';
+import TextIndentIncrease from '@spectrum-icons/workflow/TextIndentIncrease';
+import TextItalics from '@spectrum-icons/workflow/TextItalic';
+import {ToggleButton} from '@adobe/react-spectrum';
+
+export default {
+  ...defaultConfig,
+  title: 'MenuTrigger/SubMenu'
+};
+
+// TODO: add chromatic stories
+export const SubMenuStatic = {
+  render: (args) => (
+    renderMenuTrigger(
+      <Menu onAction={action('lvl 1 menu onAction')} {...args}>
+        <Item key="Lvl 1 Item 1">Lvl 1 Item 1</Item>
+        <SubMenuTrigger>
+          <Item key="Lvl 1 Item 2">Lvl 1 Item 2</Item>
+          <Menu onAction={action('lvl 2 menu onAction')} {...args.subMenu1Props}>
+            <Item key="Lvl 2 Item 1">Lvl 2 Item 1</Item>
+            <Item key="Lvl 2 Item 2">Lvl 2 Item 2</Item>
+            <SubMenuTrigger>
+              <Item key="Lvl 2 Item 3">Lvl 2 Item 3</Item>
+              <Menu onAction={action('lvl 3 menu onAction')} {...args.subMenu2Props}>
+                <Item key="Lvl 3 Item 1">Lvl 3 Item 1</Item>
+                <Item key="Lvl 3 Item 2">Lvl 3 Item 2</Item>
+                <Item key="Lvl 3 Item 3">Lvl 3 Item 3</Item>
+              </Menu>
+            </SubMenuTrigger>
+          </Menu>
+        </SubMenuTrigger>
+        <Item key="Lvl 1 Item 3">Lvl 1 Item 3</Item>
+      </Menu>
+    , args.menuTriggerProps)
+  ),
+  name: 'static submenu items'
+};
+
+let dynamicSubMenu = [
+  {name: 'Lvl 1 Item 1'},
+  {name: 'Lvl 1 Item 2', children: [
+    {name: 'Lvl 2 Item 1'},
+    {name: 'Lvl 2 Item 2'},
+    {name: 'Lvl 2 Item 3', children: [
+      {name: 'Lvl 3 Item 1'},
+      {name: 'Lvl 3 Item 2'},
+      {name: 'Lvl 3 Item 3'}
+    ]}
+  ]},
+  {name: 'Lvl 1 Item 3'}
+];
+
+// TODO: is this really the only way to define the dynamic render func? Kinda annoying that we have
+// to define this func separately and recursively call it, but I don't think we can do the same approach as
+// collapsible rows tableview since Menu isn't a collection component nor are we making the parent Menu tree collection
+// track the full "tree" of menu items (we aren't making Item take other items/sections as children and are relying on each submenu having its own
+// tree state + collection tracking its own level of items).
+let dynamicRenderFunc = (item) => {
+  if (item.children) {
+    return (
+      <SubMenuTrigger>
+        <Item key={item.name}>{item.name}</Item>
+        <Menu items={item.children} onAction={action(`${item.name} onAction`)}>
+          {(item) => dynamicRenderFunc(item)}
+        </Menu>
+      </SubMenuTrigger>
+    );
+  } else {
+    return <Item key={item.name}>{item.name}</Item>;
+  }
+};
```

Comment:
Perhaps we can provide these render funcs (see line 194-214 as well) in the docs so users don't have to come up with this themselves